File: sdk/java/src/test/java/com/flagent/client/model/PutDistributionsRequestTest.java
package com.flagent.client.model;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for PutDistributionsRequest
 */
class PutDistributionsRequestTest {

    @Test
    void testPutDistributionsRequest() {
        DistributionRequest d = new DistributionRequest().variantID(1L).variantKey("control").percent(100L);
        List<DistributionRequest> distributions = Collections.singletonList(d);
        PutDistributionsRequest model = new PutDistributionsRequest().distributions(distributions);
        Assertions.assertNotNull(model);
        Assertions.assertEquals(1, model.getDistributions().size());
        Assertions.assertEquals(100L, model.getDistributions().get(0).getPercent());
    }

    @Test
    void distributionsTest() {
        List<DistributionRequest> list = Arrays.asList(
            new DistributionRequest().variantID(1L).percent(50L),
            new DistributionRequest().variantID(2L).percent(50L)
        );
        PutDistributionsRequest model = new PutDistributionsRequest().distributions(list);
        Assertions.assertEquals(2, model.getDistributions().size());
    }

}

File: sdk/dart/test/put_distributions_request_test.dart
import 'package:built_collection/built_collection.dart';
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

// tests for PutDistributionsRequest
void main() {
  final instance = PutDistributionsRequest((b) => b
    ..distributions = ListBuilder<DistributionRequest>([
      DistributionRequest((d) => d
        ..variantID = 1
        ..variantKey = 'control'
        ..percent = 50),
      DistributionRequest((d) => d
        ..variantID = 2
        ..variantKey = 'treatment'
        ..percent = 50),
    ]));

  group(PutDistributionsRequest, () {
    test('to test the property `distributions`', () async {
      expect(instance.distributions.length, equals(2));
      expect(instance.distributions[0].variantKey, equals('control'));
      expect(instance.distributions[1].variantKey, equals('treatment'));
    });
  });
}
